src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php

```diff
@@ -326,61 +331,85 @@ protected function getCategoryListRepresentation(Request $request, $parentKey =
         $listBuilder->addSelectField($fieldDescriptors['lft']);
         $listBuilder->addSelectField($fieldDescriptors['rgt']);
 
-        if ($parentKey !== null) {
-            $this->addParentSelector($parentKey, $listBuilder);
+        return $listBuilder;
+    }
+
+    /**
+     * Expands all paths from the given root-level the the given leafs.
+     * All categories which are children of a category along this path are returned.
+     * If a given leaf is not a descendant of the given root-level, it is ignored.
+     *
+     * @param $leafIds array Ids of the categories of the leaves
+     * @param $rootIds array Ids of the categories of the root level
+     * @param $locale
+     *
+     * @return array|mixed
+     *
+     * @throws InvalidExpressionArgumentException
+     */
+    private function expandChildren($leafIds, $rootIds, $locale)
+    {
+        $leaves = $this->getManager()->findByIds($leafIds, $locale);
```

Comment:
`$leaveCategories`, and is that really needed at all? Can't we load these categories in a more efficient way?